### DISCUSSION TITLE
roguelike纯六星优选策略

### DISCUSSION THREAD

**Discussion by WarmLight7** [MaaAssistantArknights/MaaAssistantArknights#1167] [Ideas]
目测感觉每次拿一堆4星5星反而降低了maa的上限，希望新增rog mod 3只拿6星和3星，
就是在RoguelikeRecruitImageAnalyzer.cpp的analyze()函数中，新增一个条件判断，在info.level不是6或3的时候不将info emplace_back
![image](https://user-images.githubusercontent.com/59765294/178885189-a4731f5a-cf97-45a2-923e-41fd1e1d4ed6.png)

这样可能会导致的问题是analyze返回结果为空，目前还没找到对这个为空的处理是什么，如果是单纯跳过的话，我感觉即使不招募也比招募到一堆四星五星效果好一点

**Comment by MistEO** [comment]
似乎确实可以，之后可能发个测试版看看效果咋样

**Reply by WarmLight7** [reply]
我用修改json的方法试了一下，结果就是要么第一层第二层都过不了，要么能打到傀影，如果想追上限的话这么改还是挺好的，如果特化一下傀猫猫的打法，我感觉可以凹一下通关
不知道横向拖动好不好做，不拿几个三星顶着直接拿六星就容易过不了前期，但是如果拿了挺多三星也会导致后面maa不一定会不会换干员QAQ

**Reply by zzyyyl** [reply]
换干员现在好像还没有
但是可以加一个 `快捷编队-清空选择-确认-快捷编队-选人-确认` 的逻辑，这样能保证干员充足的时候不会拿很多低星干员（大概

**Reply by MistEO** [reply]
这样有排序吗？

---

哦是退出去重进的，好主意啊。我还打算识别谁是等级高的谁是等级低的。那这样就省事了

**Reply by MistEO** [reply]
> 这样有排序吗？
> 
> ---
> 
> 哦是退出去重进的，好主意啊。我还打算识别谁是等级高的谁是等级低的。那这样就省事了

v4.1.0-beta.4 fixed
